### PR TITLE
add steps api functions

### PR DESCRIPTION
**Pull request by tech4GT**
@jywarren Added all functions🎉 
Can you please elaborate on listeners.
Signed-off-by: tech4GT <varun.gupta1798@gmail.com>

### CODE REVIEW

File: src/util/getStep.js [hunk 1]
```diff
@@ -1,18 +1,54 @@
...
+            if(options[key]) options[key] = optionsObj[key];
+        }
+    },
+
+    getFormat : function(){
+        return this.getStep(-1).output.format;
+    },
+
+    getHeight : function(callback){
+        getPixels(this.getStep(-1).output.src,function(err,pixels){
```
Comment (jywarren) [inline src/util/getStep.js]: would using `getPixels` require parsing the whole image? I wonder if that'd be an expensive thing to do and if there are any reasonable alternatives. What do you think?

**Comment by jywarren** [comment]
Linking to #242 

Very cool!

File: src/util/getStep.js (hunk 1 above)
Comment (tech4GT) [reply src/util/getStep.js]: Hmm...yeah now that you have mentioned this, yes I'll change this!! Let's get the Module requirements PR merged in and rebase this before making any more changes, then we can bump a major version!!

**Comment by tech4GT** [comment]
@jywarren merged with master!!

**Comment by jywarren** [comment]
This looks great. Can we add some tests to demonstrate usage of these? Thank you, great work!

**Comment by tech4GT** [comment]
@jywarren Working on this now!!

**Comment by tech4GT** [comment]
@jywarren Actually most of these are functions which are scoped inside the module... So I would either have to write a test module which uses all of these or we can add tests as we develop modules which use these, which way should we go?

**Comment by jywarren** [comment]
lets do a test module! i think thatll be worth it :-)

**Comment by tech4GT** [comment]
@jywarren Working on this now!!

**Comment by tech4GT** [comment]
@jywarren I got an idea while working on this!! Should we implement the importModule functionality first? I mean like we discussed in the dynamic modules idea, that way we won't have to bundle this test module with the source code, we can just register it inside the tests file and run the sequencer with it(we will keep this module inside the test folder as well)

**Comment by tech4GT** [comment]
This makes a lot of sense to me since for the implementation of met-modules I have in mind this will be a welcome functionality!! Please see the meta-modules I made a flow over there...

**Comment by tech4GT** [comment]
And sorry we are postponing meta-modules again and again but I think once we are done with dynamic modules core(not completely, like we can include installing from npm part later) it would help with meta-modules since *meta-modules are themselves dynamic*

**Comment by jywarren** [comment]
Oh, I like that. Good idea!

On Jun 19, 2018 10:05 AM, "Varun Gupta" <notifications@github.com> wrote:

@jywarren <https://github.com/jywarren> I got an idea while working on
this!! Should we implement the importModule functionality first? I mean
like we discussed in the dynamic modules idea, that way we won't have to
bundle this test module with the source code, we can just register it
inside the tests file and run the sequencer with it(we will keep this
module inside the test folder as well)

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/publiclab/image-sequencer/pull/293#issuecomment-398410686>,
or mute the thread

**Comment by tech4GT** [comment]
@jywarren So Should we merge this in right now or rebase this over dynamic modules pr?

**Comment by jywarren** [comment]
Let's delay this until we can include the dynamic test module. Is that ok?

**Comment by tech4GT** [comment]
@jywarren Sure!! So now I am doing this in a way that just by adding a few lines to the existing module file it can be made dynamically loadable!!

**Comment by jywarren** [comment]
Awesome

**Comment by tech4GT** [comment]
@jywarren This is ready too!!🎉

File: test/modules/testModule/Module.js [hunk 2]
```diff
@@ -1,14 +1,27 @@
...
-    var output = function(input) {
-      return input;
+    var step = this;
+
+    this.getPreviousStep();
+    this.getNextStep();
+    this.getInput(1);
+    this.getOutput(1);
+    this.getOptions();
+    this.getFormat();
```
Comment (jywarren) [inline test/modules/testModule/Module.js]: Can we assert anything with these?

Comment (tech4GT) [reply test/modules/testModule/Module.js]: @jywarren I thought we just wanted to check if they throw an error or not, actually som of these like height I can hardcode at best, will that be ok?

**Comment by tech4GT** [comment]
@jywarren Done!!

**Comment by tech4GT** [comment]
@jywarren Please merge this in. Thanks

**Comment by jywarren** [comment]
🎉

**Comment by tech4GT** [comment]
Awesome!!!

**Pull request merged**